syntax = "proto3";

enum SelectOption {
    GET = 0;
    GET_ALL = 1;
    ADD = 2;
    REGISTER=3;
    LOGIN=4;
    CURRENT_USER=5;
    REFRESH_TOKEN=6;
    CREATE = 7;
    UPDATE =8;
    REDUCE = 9;
    DELETE =10;
}


message User {
    int32 id =1;
    string user_id=2;
    string username =3;
    string email=4;
    string password =5;
    string access_token = 6;
    string refresh_token = 7;
    string error_message = 8;
    SelectOption option =9;
    int64 http_status_code = 10;
}


message Order {
    int32 id =1;
    string order_id=2;
    string product_id =3;
    int64 quantity= 4;
    string shipping_address = 5;
    string customer_notes = 6;
    int64 stock_level = 7;
    bool is_product_available = 8;
    bool is_stock_available  = 9;
    string message = 10;
    string error_message = 11;
    SelectOption option =12;
    int64 http_status_code = 13;
}

message OrderList {
    repeated Order orders = 1;
}